packages/aws-cdk-lib/aws-sns/lib/policy.ts

```diff
@@ -51,9 +60,31 @@ export class TopicPolicy extends Resource {
 
     this.document = props.policyDocument ?? this.document;
 
+    if (props.enforceSsl) {
+      props.topics.map(t => this.document.addStatements(this.createSslPolicyDocument(t.topicArn)));
+    }
+
     new CfnTopicPolicy(this, 'Resource', {
       policyDocument: this.document,
       topics: props.topics.map(t => t.topicArn),
     });
   }
+
+  /**
+   * Adds a statement to enforce encryption of data in transit when publishing to the topic.
+   *
+   * For more information, see https://docs.aws.amazon.com/sns/latest/dg/sns-security-best-practices.html#enforce-encryption-data-in-transit.
+   */
+  protected createSslPolicyDocument(topicArn: string): PolicyStatement {
+    return new PolicyStatement ({
+      sid: 'AllowPublishThroughSSLOnly',
+      actions: ['sns:Publish'],
+      effect: Effect.DENY,
+      resources: [topicArn],
+      conditions: {
+        Bool: { 'aws:SecureTransport': 'false' },
+      },
+      principals: [new AnyPrincipal()],
```

Comment:
The SNS docs go back and forth but I _think_ this should be `AnyPrincipal` not `StarPrincipal`.

Reply:
I changed this to star to match this specific [doc](https://docs.aws.amazon.com/sns/latest/dg/sns-security-best-practices.html#enforce-encryption-data-in-transit).